modules/standard/Path.chpl

```diff
@@ -457,4 +457,78 @@ proc file.realPath(): string throws {
      return ("", name);
    }
  }
+
+/* Returns the argument.  If the argument begins with `~` or `~<user>`, replace
+   that substring with that user's home directory, otherwise returns the
+   argument unchanged.
+
+   :arg path: a file or directory, which may or may not include `~` or `~<user>`
+   :type path: `string`
+
+   :return: the argument, replacing any `~` or `~<user>` with the user's home
+                directory
+   :rtype: `string`
+*/
+proc expandUser(path: string) : string
+  {
+    var i: int = 1;
+    var j: int = 1;
+    var k: int = 1;
+    var lenz: int;
+    var lenp: int = path.length;;
+    var count: int = 0;
+    var x: c_string ;
+    var y: c_string;
```

Comment:
Similarly, `x` and `y` could be declared on the same line:
```chapel
var x, y: c_string;
```

Reply:
However, I would instead recommend declaring them closer to where they are used (inside the `if` statement), and giving `x` the value of `"PWD"` when you declare it.

Reply:
It would be good to give these variables more descriptive names than `x`, `y`, or `z`.  This will make it easier to maintain or understand the code.  Iteration variables like `i` are fine because they are generally understood to be for that purpose, though since there are so many of them it might be good to give them more precise names as well.